### PR TITLE
Add network fine tuning examples

### PR DESCRIPTION
**Pull request by futurely**
These examples show how to only train the last fully connected layer of a network with all the other parameters loaded from a pre-trained model.

Related issues include https://github.com/dmlc/mxnet/issues/1313, https://github.com/dmlc/mxnet/issues/1340,  https://github.com/dmlc/mxnet/issues/1439, and https://github.com/dmlc/mxnet/pull/1568.

### CODE REVIEW

**Comment by winstywang** [comment]
Using BlockGrad is not a proper choice, since the weight decay will still affect the weights. So the best practice is to use a customer optimizer. You can set the learning rate of the fixed layers to 0. This is also more flexible. You can set the learning rate of previous layer to 0.1 of the base one.

**Comment by tqchen** [comment]
Close as per @winstywang 's comment, please open another PR if we have a new solution with learning rate set to 0. Thanks!

**Comment by EasonD3** [comment]
Any follow-ups so far?

**Comment by antinucleon** [comment]
You can use following code from GraphLab Create:
```
def get_feature_symbol(model, top_layer=None):
    """Get feature symbol from a model
    .. note::
        If top_layer is not present, it will return the second last layer symbol
    Parameters
    ----------
    model: mx.model.FeedForward
        Model will be used to extract feature symbol
    top_layer: str, option
        Name of top_layer will be used
    Returns
    -------
    internals[top_layer]: mx.symbol.Symbol
        Feature symbol
    """
    internals = model.symbol.get_internals()
    tmp = internals.list_outputs()[::-1]
    outputs = [name for name in tmp if name.endswith("output")]
    if top_layer != None and type(top_layer) != str:
        error_msg = "top_layer must be a string in following candidates:\n %s" % "\n".join(outputs)
        raise TypeError(error_msg)
    if top_layer == None:
        assert len(outputs) > 3
        top_layer = outputs[2]
    else:
        if top_layer not in outputs:
            error_msg = "%s not exists in symbol. Possible choice:\n%s" \
                    % (top_layer, "\n".join(outputs))
            raise ValueError(error_msg)
    return internals[top_layer]

def finetune(symbol, model, **kwargs):
    """Get a FeedForward model for fine-tune
    .. note::
        For layer doesn't exist in model, will be initialized as uniform random weight
    Parameters
    ----------
    symbol: mx.symbol.Symbol
        Symbol of new network will be finetuned.
    model: mx.model.FeedForward
        Model which contains parameters which will be used for fine-tune.
    kwargs: kwargs
        mx.model.create function's parameters
    Returns
    -------
    new_model: mx.model.FeedForward
        Finetuned model
    Examples
    --------
    Load a model
    >>> sym, arg_params, aux_params = mx.model.load_checkpoint("model", 9)
    Make new symbol for finetune
    >>> feature = mx.model.get_feature_symbol(model)
    >>> net = mx.sym.FullyConnected(data=feature, num_hidden=10, name="new_fc")
    >>> net = mx.sym.SoftmaxOutput(data=net, name="softmax")
    Finetune the model
    >>> new_model = mx.model.finetune(symbol=net, model=model, num_epoch=2, learning_rate=1e-3,
                                      X=train, eval_data=val,
                                      batch_end_callback=mx.callback.Speedometer(100))
    """
    initializer = Load(param=model.arg_params, default_init=Uniform(0.001))
    new_model = FeedForward.create(symbol=symbol, initializer=initializer, **kwargs)
    return new_model
```
And pass new optimizer in to set different learning rate

**Comment by EasonD3** [comment]
Sweet. Thanks! @antinucleon
